com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -1102,5 +1129,10 @@ internal uint HostCheckForGlobalObjectIdHashOverride()
 
             return GlobalObjectIdHash;
         }
+
+        public void NetworkSerialize<T>(BufferSerializer<T> serializer) where T : IBufferSerializerImplementation
+        {
+            serializer.SerializeValue(ref m_NetworkObjectId);
```

Comment:
This doesn't really work as we are not technically serializing the NetworkObject here. From what I can tell it is also unused. I suggest that we remove this because users could try to put this into an RPC since it supports INetworkSerializable which would be a bad idea.

I created [this](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/1173) instead for NetworkObject serialization. What do you think?

Reply:
oh... yeah, this was a flawed first attempt that I meant to delete. NetworkObject serialization is done through FastBufferWriterExtensions and FastBufferReaderExtensions (which do, in fact, allow NetworkObject to be sent through an RPC). You are correct this is unused.

Reply:
yeah, I'm with Luke here. let's not do this and don't confuse users about what's actually going on here. that other linked PR is crafted specifically for this type of linking/referencing :)

Reply:
Could I convince you to get rid of these extension methods and use the approach I suggested in my PR instead. I think having serialization for GameObject & NetworkObject like this is bad idea because of the same arguments I mentioned in the previous PR:
- It relies on the singleton NetworkManager.
- It heavily relies on perfectly ordered spawn, RPC and NetworkVariable messages. If something were to arrive unordered before a spawn message the deserialization would not find the object.

Reply:
Done.